platform/ui/src/contextProviders/getPresentationId.ts

```diff
@@ -0,0 +1,73 @@
+/**
+ * Selects a presentation ID to use for this viewport.
+ * This is done to allow the same display set to be displayed more than once
+ * on screen, with different attributes such as window level and initial position.
+ * Then, when redisplaying that, the nearest/most common attribute is re-used.
+ *
+ * For example, for display set <displaySetUID>, in a viewport of type volume,
+ * the generated presentationID might be
+ * `volume:axial:<displaySetUID>`.  This can then be used to store and retrieve
+ * presentation information in state sync service 'presentationSync' state.
+ *
+ * The generated value attempts to generate a unique value for every type
+ * of viewport which should have it's own presentation information.  Thus, the
+ * following values are used for presentation ID:
+ *
+ * 1. viewportType - since the presentation information for a volume is different than for a stack
+ * 2. orientation - since the camera is different for different orientations
+ * 3. display set instance UID - since different display sets should get displayed differently
+ * 4. instance count - since displaying the same series twice should allow applying different window level etc
+ *
+ * @param viewport requiring a presentation Id
+ * @param viewports is the list of viewports.  Newly assigned viewports
+ *    have the presentation Id cleared, while those remaining or those already
+ *    assigned presentation ID's have the presentation ID remaining.
```

Comment:
I am having a difficult time understanding "while those remaining or those already assigned presentation ID's have the presentation ID remaining". In particular I do not understand what it means for a viewport to "have the presentation ID remaining". Pleasee clarify.